src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -639,6 +626,63 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, Diagnostic
             }
         }
 
+        private BoundExpression BindTupleExpression(TupleExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            var arguments = node.Arguments;
+            if (arguments.Count < 2)
+            {
+                // this should be a parse error already.
+                return BadExpression(node);
+            }
+
+            var boundArguments = ArrayBuilder<BoundExpression>.GetInstance(arguments.Count);
+            var elementTypes = ArrayBuilder<TypeSymbol>.GetInstance(arguments.Count);
+            ArrayBuilder<string> elementNames = null;
+
+            for (int i = 0, l = arguments.Count; i < l; i++)
+            {
+                var argumentSyntax = arguments[i];
+                var name = argumentSyntax.NameColon?.Name.Identifier.ValueText;
+
+                // names would typically all be there or none at all
+                // but in case we need to handle this in error cases
+                if (elementNames != null)
+                {
+                    elementNames.Add(name ?? "Item" + i);
+                }
+                else
+                {
+                    if (name != null)
+                    {
+                        elementNames = ArrayBuilder<string>.GetInstance(arguments.Count);
+                        for (int j = 0; j < i; j++)
+                        {
+                            elementNames.Add(name ?? "Item" + j);
+                        }
+                        elementNames.Add(name);
+                    }
+                }
+
+                var boundArgument = BindExpression(argumentSyntax.Expression, diagnostics);
+                boundArguments.Add(boundArgument);
+                elementTypes.Add(boundArgument.Type);
+            }
+
+            var type = new TupleTypeSymbol(
+                elementTypes.ToImmutableAndFree(),
+                elementNames == null ?
+                                ImmutableArray<string>.Empty :
+                                elementNames.ToImmutableAndFree(),
+                node,
+                this,
+                diagnostics);
+
```

Comment:
it looks like symbol construction expects empty, but that can be changed to default if needed